community/detectors/spring_framework_cve_2022_22965/src/main/java/com/google/tsunami/plugins/detectors/spring/SpringCve202222965Detector.java

```diff
@@ -0,0 +1,220 @@
+        simpleCrawler.crawlAsync(crawlConfig);
+    ImmutableSet<CrawlResult> crawlResults = null;
+    try {
+      crawlResults = crawlResultsFuture.get();
+    } catch (Exception e) {
+      logger.atWarning().withCause(e).log("Unable to crawl.");
+      return false;
+    }
+    for (CrawlResult crawlResult : crawlResults) {
+      String targetUri = crawlResult.getCrawlTarget().getUrl();
+      String httpMethod = crawlResult.getCrawlTarget().getHttpMethod();
+      if (proofOfConcept(targetUri, httpMethod, networkService)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private boolean proofOfConcept(String targetUri,
+      String httpMethod, NetworkService networkService) {
+    HttpResponse firstPoCResponse;
+    HttpResponse secondPoCResponse;
+    try {
+      switch (httpMethod) {
+        case "GET":
+          firstPoCResponse = httpClient.send(get(targetUri+"?"+VULNERABILITY_PAYLOAD_STRING_1)
+                  .withEmptyHeaders()
+                  .build(),
+              networkService
+          );
+          secondPoCResponse = httpClient.send(get(targetUri+"?"+VULNERABILITY_PAYLOAD_STRING_2)
+                  .withEmptyHeaders()
+                  .build(),
+              networkService
+          );
+          break;
+        case "POST":
+          firstPoCResponse = httpClient.send(post(targetUri+"?"+VULNERABILITY_PAYLOAD_STRING_1)
+                  .withEmptyHeaders()
+                  .build(),
+              networkService
+          );
+          secondPoCResponse = httpClient.send(post(targetUri+"?"+VULNERABILITY_PAYLOAD_STRING_2)
+                  .withEmptyHeaders()
+                  .build(),
+              networkService
+          );
+          break;
+        default:
+          logger.atWarning().log("Unable to query '%s'.", targetUri);
+          return false;
+      }
+      if (firstPoCResponse.status() == HttpStatus.OK
+          && secondPoCResponse.status() == HttpStatus.BAD_REQUEST) {
+        return true;
+      }
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Unable to query '%s'.", targetUri);
+    }
+    return false;
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(VulnerabilityId.newBuilder().setPublisher("TSUNAMI_COMMUNITY")
+                    .setValue("CVE_2022_22965"))
+                .setSeverity(Severity.CRITICAL)
+                .setTitle("Spring Framework RCE CVE-2022-22965")
+                .setDescription("A Spring MVC or Spring WebFlux application running on JDK"
```

Comment:
Can you please also set the `recommendation` field so that the user receiving the report knows how to fix the issue?